packages/schema/src/model.ts

```diff
@@ -538,8 +559,6 @@ export class JupyterGISModel implements IJupyterGISModel {
         this._removeLayerTreeLayer(currentItem.layers, layerIdToRemove);
       }
     }
-
-    this.sharedModel.layerTree = layerTree;
```

Comment:
I wonder if this is not the main culprit, and if all the other changes are necessary.

If I'm not mistaken, this was called in a recursive function on the `layerTree`, which means that the `sharedModel.layerTree` was updated with wrong values. When it was processing a nested group, at the end of the function, the `layerTree` become this group temporary. Then it finalizes the process of the parent group, which also set the `layerTree` to this parent group, etc... 
`this.sharedModel.layerTree = layerTree` should be called only when all the nested groups have been processed, I think.

Reply:
Tested successfully with https://github.com/geojupyter/jupytergis/pull/269, I can't reproduce [this bug](https://github.com/geojupyter/jupytergis/pull/269#pullrequestreview-2527185205) with a very few change (add an intermediate function):

```ts
 private _removeLayerTreeGroup(
    layerTree: IJGISLayerItem[],
    groupName: string
  ) {
    this._removeLayerTreeGroupRecursive(layerTree, groupName);
    this.sharedModel.layerTree = layerTree;
  }

  private _removeLayerTreeGroupRecursive(
    layerTree: IJGISLayerItem[],
    groupName: string
  ) {

    // Same as before except that I removed the last line that update the layerTree

}
```

And the same to remove a layer.